src/hexkit/providers/s3/testutils/_fixtures.py

```diff
@@ -111,19 +110,66 @@ async def populate_file_objects(self, file_objects: list[FileObject]):
             self.storage, bucket_fixtures=[], object_fixtures=file_objects
         )
 
-    async def empty_buckets(self, buckets_to_exclude: Optional[list[str]] = None):
-        """Clean the test artifacts or files from the populated buckets."""
-        for bucket in self.get_buckets().difference(buckets_to_exclude or []):
+    async def empty_buckets(
+        self,
+        *,
+        buckets: Optional[Union[str, list[str]]] = None,
+        exclude_buckets: Optional[list[str]] = None,
+    ):
+        """Remove all test objects from the given bucket(s).
+
+        If no buckets are specified, all existing buckets will be emptied,
+        therefore we recommend to always specify a bucket list.
+        You can also specify bucket(s) that should be excluded
+        from the operation, i.e. buckets that should not be emptied.
+        """
+        if buckets is None:
+            buckets = self.get_buckets()
+        elif isinstance(buckets, str):
+            buckets = [buckets]
+        if exclude_buckets is None:
+            exclude_buckets = []
+        elif isinstance(exclude_buckets, str):
+            exclude_buckets = [exclude_buckets]
+        excluded_buckets = set(exclude_buckets)
+        for bucket in buckets:
+            if bucket in excluded_buckets:
+                continue
             # Get list of all objects in the bucket
             object_ids = await self.storage.list_all_object_ids(bucket_id=bucket)
             # Delete all of these objects
             for object_id in object_ids:
+                # Note that id validation errors can be raised here if the bucket
+                # was populated by other means than the S3 storage fixture.
+                # We intentionally do not catch these errors because they
+                # usually mean that you're operating on the wrong buckets.
                 await self.storage.delete_object(bucket_id=bucket, object_id=object_id)
 
-    async def delete_buckets(self, buckets_to_exclude: Optional[list[str]] = None):
-        """Delete the populated buckets."""
-        for bucket in self.get_buckets().difference(buckets_to_exclude or []):
-            await self.storage.delete_bucket(bucket, delete_content=True)
+    async def delete_buckets(
+        self,
+        *,
+        buckets: Optional[Union[str, list[str]]] = None,
+        exclude_buckets: Optional[list[str]] = None,
+    ):
+        """Delete the given bucket(s).
+
+        If no buckets are specified, all existing buckets will be deleted,
+        therefore we recommend to always specify a bucket list.
```

Comment:
If there's really a threat that buckets outside of test could be unintentionally deleted, maybe it would be better to mandate a bucket list every time.

Reply:
In our usual environment (integration tests using test containers) this is not a problem. So I would not want to make it too restrictive.